### PR TITLE
Fix sync failure when API returns a null when an empty list is expected

### PR DESCRIPTION
**Pull request by simonrozsival**
Ref #545 but does not close it - this is only a fix of s sub-problem.

This happens quite a lot when user has either no time entry at all or when nothing changes between two "get since" API calls.

Squash whenever.

### CODE REVIEW

File: Toggl.Foundation.Tests/Sync/States/Pull/PersistStateTests.cs [hunk 1]
```diff
@@ -171,6 +177,18 @@ public void UpdatesSinceParametersOfTheFetchedEntity()
                     newSinceParameters => OtherSinceDatesDidntChange(oldSinceParameters, newSinceParameters, newAt)));
             }
 
+            public void HandlesNullValueReceivedFromTheServerAsAnEmptyList()
+            {
+                var oldSinceParameters = new SinceParameters(null);
+                var entities = (List<TInterface>)null;
```
Comment (willsb) [inline Toggl.Foundation.Tests/Sync/States/Pull/PersistStateTests.cs]: I think `List<TInterface> entities = null` is a lot clearer

**Comment by simonrozsival** [comment]
I don't think that that was caused by deserialization - I tried putting the JSON you sent me into deserialization tests and it worked just fine. But I will try to simulate it again.

**Comment by simonrozsival** [comment]
Could you please create a separate issue for the deserialization error and add it to Sprint III? I think it would be better to deal with it in a separate PR once this is resolved.

**Comment by willsb** [comment]
Yeah, I definitely think that it should be handled by another PR. I'm just wondering if it's an issue or not. You just said that the error didn't happen when you tried the faulty json. Should I open the issue or not?

**Comment by simonrozsival** [comment]
I think we should be sure that there are no errors and if there was some problem which we can't identify, then I think it is an issue. The worst thing that can happen is that I won't be able to reproduce it anyhow and then we will close it until it surfaces again.

**Pull request merged**